tests/performance/test_performance_benchmarks.py

```
@@ -0,0 +1,700 @@
+        assert duration < 20.0  # Should complete within 20 seconds
+
+
+@pytest.mark.asyncio
+class TestMemoryAndResourceUsage:
+    """Tests for memory usage and resource management."""
+
+    async def test_memory_usage_with_large_datasets(self, large_document_dataset):
+        """Test memory usage patterns with large datasets."""
+        import os
+
+        import psutil
```

Pylint (3.3.7)

[convention] Import outside toplevel (psutil)

(C0415)